File: src/dev/4.5/Legacy.EFLViewModelBase/Attributes/ModelPropertyAttributes.vb
Imports System.Xml.Serialization

<AttributeUsage(AttributeTargets.Property)>
Public Class ModelPropertyIgnoreAttribute
    Inherits Attribute

End Class

<AttributeUsage(AttributeTargets.Class)>
Public Class ModelPropertiesCompleteAttribute
    Inherits Attribute

End Class

<AttributeUsage(AttributeTargets.Property)>
Public Class ModelPropertyNameAttribute
    Inherits Attribute

    Sub New(propertyName As String)
        Me.PropertyName = propertyName
    End Sub

    Sub New(propertyName As String, mustExist As Boolean)
        Me.PropertyName = propertyName
        Me.MustExist = mustExist
    End Sub

    Public Property PropertyName As String
    Public Property MustExist As Boolean

End Class

File: src/dev/MvvmBaseLib/MvvmBaseLibShared/MvvmBaseClasses/ModelPropertyAttributes.cs
using System;

namespace ActiveDevelop.MvvmBaseLib.Mvvm
{
    [AttributeUsage(AttributeTargets.Property)]
    public class ModelPropertyIgnoreAttribute : Attribute
    {
    }

    [AttributeUsage(AttributeTargets.Class)]
    public class ModelPropertiesCompleteAttribute : Attribute
    {
    }

    [AttributeUsage(AttributeTargets.Property)]
    public class ModelPropertyNameAttribute : Attribute
    {
        public ModelPropertyNameAttribute(string propertyName)
        {
            this.PropertyName = propertyName;
        }

        public ModelPropertyNameAttribute(string propertyName, bool mustExist)
        {
            this.PropertyName = propertyName;
            this.MustExist = mustExist;
        }

        public string PropertyName { get; set; }
        public bool MustExist { get; set; }

    }
}